CONTRIBUTING.md

```diff
@@ -14,16 +14,30 @@ possible.
  
    You can also run the formatters manually as follows:
  
-     ```
+     ```bash
      clang-format -i file.cpp
      ```
  
-     ```
+     ```bash
      black file.py
      ```
  
    or run `pre-commit run --all-files` to check all files in the repo.
 
+## Tips on porting LLMs from HuggingFace
+
+First make sure model you want to port has weights in [safetensor](https://huggingface.co/docs/safetensors/index) format, if not you can use convert using [this](https://huggingface.co/spaces/safetensors/convert) tool.
+
+After that it's relatively simple, you need to add the model file to `llms/mlx_lm/models` directory 
+
+You can see other examples in there, we recommend starting from one of the models that is pretty similar to model you want to add, as most models are based on LLAMA/Mistral architechture.
+
+Make sure the name of the model file is the same as the `model_type` in the `config` for example [starcoder2](https://huggingface.co/bigcode/starcoder2-7b/blob/main/config.json#L17) 
+
+For model layer names we suggest either referring to the transformers implementation if you are familiar with the codebase, or loading the model weights and checking the weight names which will give you a hint of the model structure, you can take a look at the names of the weights by inspecting `model.safetensors.index.json`
+
+If you want to add LoRA support you can edit [`mlx_lm/tuner/utils.py`](https://github.com/ml-explore/mlx-examples/blob/main/llms/mlx_lm/tuner/utils.py#L27-L60)
+
```

Comment:
Thanks! I think it's great to add something like  this but let's add it in the MLX LM docs rather than the general contributing guidelines. Maybe in a CONTRIBUTING.md in that subdirectory?

Reply:
@awni done